clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -560,12 +561,14 @@ private List<ProducerBatch> drainBatchesForOneNode(Cluster cluster, Node node, i
         List<PartitionInfo> parts = cluster.partitionsForNode(node.id());
         List<ProducerBatch> ready = new ArrayList<>();
         /* to make starvation less likely this loop doesn't start at 0 */
+        int drainIndex = getDrainIndex(node.idString());
         int start = drainIndex = drainIndex % parts.size();
+        updateDrainIndex(node.idString(),drainIndex);
```

Comment:
I think we can update Drain Index at the end of `drainBatchesForOneNode`. We don't need to update it each time.